docs/source/notebooks/Boson_Bunching.ipynb

```diff
@@ -0,0 +1,1076 @@
+    "circuit_rand = pcvl.Circuit.decomposition(unitary, mzi,\n",
+    "                                               phase_shifter_fn=pcvl.PS,\n",
+    "                                               shape=\"triangle\")\n",
+    "\n",
+    "pcvl.pdisplay(circuit_rand, recursive=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Inputting photons of different degrees of distinguishability\n",
+    "\n",
+    "Now that we have the circuit which represents the correct unitary matrix and therefore the correct $H$ from the conjecture, we need to feed in the right photons following the distinguishably matrix $S=A$. In the [paper](http://arxiv.org/abs/2203.01306)[2] the input photons are described via polarization. In this simulation it is not specified how the photons are distinguishable. The first two photons are fully distinguishable. The other five photons are each a linear combination of the first two photons and form an equally spaced star shape when drawn on the Bloch sphere. \n",
+    "\n",
+    "\n",
+    "\n",
+    "![Blochsphere](../_static/img/Blochsphere.png)\n",
+    "\n",
+    "Here one can see the two fully distinguishable photons with red arrows and the five linear combinations in the equator plane."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_inputs(n,w):\n",
+    "    #make a list of fully distinguishable photons of the form |{a:i}> that means we give the photon an attribute a with value i \n",
+    "    states_dist = []\n",
+    "    for i in range(1,n+1):\n",
+    "        x = \"|{{a:{}}}>\".format(i)\n",
+    "        states_dist.append(pcvl.StateVector(x))\n",
+    "    #make a list of all input photons which are superpositions of |{a:1}> and |{a:2}>\n",
+    "    states_par_dist = [states_dist[0], states_dist[1]] #start with the pure states |{a:1}> and |{a:2}>\n",
+    "    for i in range(2,n):\n",
+    "        x = states_par_dist[0] +  states_par_dist[1] * w**(i-2) #add the states |{a:1}> + w^(i-2)|{a:2}>\n",
+    "        states_par_dist.append(x)\n",
+    "    #Input states\n",
+    "    #initialise the variables\n",
+    "    indisting_photons = [] \n",
+    "    par_disting_photons = 1 \n",
+    "    disting_photons = 1 \n",
+    "    #fill the states\n",
+    "    for i in range(n):\n",
+    "        indisting_photons.append(1) # gives the state |1,1, ... , 1>\n",
+    "        par_disting_photons = par_disting_photons * states_par_dist[i]\n",
+    "        disting_photons = disting_photons * states_dist[i] # gives the state |{a:1},{a:2}, ... ,{a:n}> \n",
+    "    return pcvl.BasicState(indisting_photons), par_disting_photons, disting_photons\n",
+    "\n",
+    "\n",
+    "input_indistinguishable, input_partiallydistinguishable, input_distinguishable = create_inputs(n,w) #indistinguishable photons, partially distinguishable photons, fully distinguishable photons respectively "
```

Comment:
In this bit of code you use a some variable that seems to correspond :
indisting_photons <=> input_indistinguishable
par_disting_photons <=> input_partiallydistinguishable
disting_photons <=> input_distinguishable
But you don't have a unity among your variable naming, I suggest using the same for each couple